vmm/src/vm.rs

```diff
@@ -820,16 +831,69 @@ impl DeviceManager {
         pci: &mut PciConfigIo,
         buses: &mut BusInfo,
         interrupt_info: &InterruptInfo,
+        mem_slots: &mut u32,
     ) -> DeviceManagerResult<()> {
         // Add virtio-fs if required
         if let Some(fs_list_cfg) = &vm_info.vm_cfg.fs {
             for fs_cfg in fs_list_cfg.iter() {
                 if let Some(fs_sock) = fs_cfg.sock.to_str() {
+                    let mut cache: Option<(VirtioSharedMemoryList, u64)> = None;
+                    if let Some(fs_cache) = fs_cfg.cache_size {
+                        // The memory needs to be 2MiB aligned in order to support
+                        // hugepages.
+                        let fs_guest_addr = allocator
+                            .allocate_mmio_addresses(
+                                None,
+                                fs_cache as GuestUsize,
+                                Some(0x0020_0000),
+                            )
+                            .ok_or(DeviceManagerError::FsRangeAllocation)?;
+
+                        let addr = unsafe {
+                            libc::mmap(
```

Comment:
Probably should look at errno.

Reply:
Yes I forgot to check the error returned here, I'm gonna fix that.